server/api/helpers/parseUserQuery/parse-user-query.js

```diff
@@ -0,0 +1,242 @@
+const xlsx = require('node-xlsx');
+
+const verifyUserData = (userData) => {
+  return (
+    // eslint-disable-next-line eqeqeq
+    userData[0] != null && userData[1] != null && userData[2] != null && userData[3] != null &&
+    userData[0] !== '' && userData[1] !== '' && userData[2] !== '' && userData[3] !== ''
+  );
+};
+
+// Format of EXCEL file
+//   |      A      |     B     |        C          |      D       |
+// 1 | lifeGroupId   | someId  |                   |              |
+// 2 | lifeGroupName | name    |                   |              |
+// 3 |               |         |                   |              |
+// 4 | Name          | Phone # | Email             | Lifestage    |
+// 5 | personA       | 123     | personA@gmail.com | Single Adult |
+const parseWorksheet = (worksheet, isBatch) => {
+  const rawData = worksheet.data;
+  // eslint-disable-next-line eqeqeq
+  if (rawData == null) return {};
+  const data = rawData.filter((e, i) => i < 5 || rawData[i].length === 4);
+
+  const lifeGroupName = isBatch ? data[1][1] : data[0][1];
+  let lifeGroupId;
+  const users = [];
+  if (isBatch) {
+    lifeGroupId = data[0][1];
+    for (let i = 4; i < data.length; i++) {
+      if (data[i].length !== 4) continue;
+
+      const userData = data[i];
+      if (verifyUserData(userData)) {
+        users.push({
+          name: userData[0],
+          phoneNumber: userData[1],
+          email: userData[2],
+          lifestage: userData[3]
+        });
+      }
+    }
+  } else {
+    const userData = data[3];
+    if (userData.length === 4 && verifyUserData(userData)) {
+      users.push({
+        name: userData[0],
+        phoneNumber: userData[1],
+        email: userData[2],
+        lifestage: userData[3]
```

Comment:
i think we said we need Campus as well? I think this will be important for DB as well as we would need that as demographic data.

Reply:
We can, but I can imagine it being very annoying for UST  leaders (or any campus leaders) to have to fill every single line with the same thing

Reply:
yeah idk what would be a better workflow, do we just force set it from the backend? like this LG is defined as under a certain campus... so all it's members are set to be part of the campus. but not sure if this is a good/complete solution. e.g. if there are people that are part of "Others" LG

Reply:
Yeah, wouldn't work in summer either. I think potential solution is just to require them to fill the campus column and also for u to create the LG directory template which also requires members to fill in their campus.

That way leaders just have to copas